pkg/controllers/deprovisioning/types.go

```diff
@@ -136,38 +136,31 @@ func (c *Candidate) lifetimeRemaining(clock clock.Clock) float64 {
 	return remaining
 }
 
-type action byte
+type Command struct {
+	candidates   []*Candidate
+	replacements []*scheduling.Machine
+}
 
 const (
-	actionDelete action = iota
-	actionReplace
-	actionDoNothing
+	noOpAction = "no-op"
```

Comment:
I think maybe these actions should also be publicly available too? Imagine we decide to split up deprovisioning into separate sub-packages down the line

Reply:
Ah, sure I don't think it's necessary since we're not using them elsewhere anyway, but I'm happy to change.